kong/plugins/rate-limiting/policies/init.lua

```diff
@@ -84,20 +84,23 @@ return {
         return nil, err
       end
 
-      if conf.redis_password and conf.redis_password ~= "" then
+      local times, err = red:get_reused_times()
+      if err then
+        ngx_log(ngx.ERR, "failed to get connect reused times: ", err)
+        return nil, err
+      end
+      if times == 0 and conf.redis_password and conf.redis_password ~= "" then
         local ok, err = red:auth(conf.redis_password)
         if not ok then
-          ngx_log(ngx.ERR, "failed to connect to Redis: ", err)
+          ngx_log(ngx.ERR, "failed to auth Redis: ", err)
           return nil, err
         end
       end
 
-      if conf.redis_database ~= nil and conf.redis_database > 0 then
-        local ok, err = red:select(conf.redis_database)
-        if not ok then
-          ngx_log(ngx.ERR, "failed to change Redis database: ", err)
-          return nil, err
-        end
+      local ok, err = red:select(conf.redis_database or 0)
```

Comment:
Could we add a comment above this line to explain why we need this? As we like to say, we do not trust our future selves to remember that this is because the redis connection pool conflicts with that of other instances of this plugin, or that of the response-ratelimiting plugin :)

Thanks!